Stetskyi_Homework_6/DataStructures/DataStructures/Tasks/DoublyLinkedList.cs

```diff
@@ -5,43 +5,118 @@
 
 namespace Tasks
 {
-    public class DoublyLinkedList<T> : IDoublyLinkedList<T>
+    public class DoublyLinkedList<T> : IDoublyLinkedList<T>, IEnumerator<T>, IEnumerator
     {
-        public int Length => throw new NotImplementedException();
+        public T[] elementArray = new T[0];
+        int position = -1;
+
+        public T this[int index]
+        {
+            get { return elementArray[index]; }
+            set { elementArray[index] = value; }
+        }
+
+        public object Current
+        {
+            get { return elementArray[position]; }
+        }
+
+        T IEnumerator<T>.Current
+        {
+            get { return elementArray[position]; }
+        }
+
+        public int Length
+        {
+            get { return elementArray.Length; }
+        }
 
         public void Add(T e)
         {
-            throw new NotImplementedException();
+            var newArray = new T[elementArray.Length + 1];
+            elementArray.CopyTo(newArray, 0);
+            newArray[newArray.Length - 1] = e;
+            elementArray = newArray;
         }
 
         public void AddAt(int index, T e)
         {
-            throw new NotImplementedException();
+            var newArray = new T[elementArray.Length + 1];
+            elementArray.CopyTo(newArray, 0);
+            for (int i = elementArray.Length - 1; i > index; i--)
+            {
+                newArray[i] = newArray[i - 1];
+            }
+            newArray[index] = e;
+            elementArray = newArray;
+        }
+
+        public void Dispose()
+        {
+            Reset();
+        }
+
+        public void Reset()
+        {
+            position = -1;
         }
 
         public T ElementAt(int index)
         {
-            throw new NotImplementedException();
+            return elementArray[index];
         }
 
         public IEnumerator<T> GetEnumerator()
         {
-            throw new NotImplementedException();
+            return this;
         }
 
         public void Remove(T item)
         {
-            throw new NotImplementedException();
+            DoublyLinkedList<T> result = new DoublyLinkedList<T>();
+            bool removed = false;
+            foreach (var element in elementArray)
+            {
+                if (element.Equals(item) && !removed)
+                {
+                    removed = true;
+                    continue;
```

Comment:
In this case, I think we have to because it helps us to skip one result.Add iteration. In this type of deletion, we will not have an empty element after deletion and our list length will be decreased by 1.
Please correct me if I am getting something wrong

<img width="309" alt="remove_Continue" src="https://user-images.githubusercontent.com/105840950/170110043-fb402bb7-1ec5-4eb6-b817-a6d84054c4ea.png">